recipes/naturalneighbor/meta.yaml

```diff
@@ -0,0 +1,46 @@
+{% set name = "naturalneighbor" %}
+{% set version = "0.2.1" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.org/packages/source/{{ name[0] }}/{{ name }}/naturalneighbor-{{ version }}.tar.gz
+  sha256: 647b5922c3787887c04be7748f8112a1720b1323c8e86b4c6665ce7eee89aee7
+
+build:
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - {{ stdlib("c") }}
+    - setuptools <65
+  host:
+    - python
+    - numpy
+    - pip
+    - setuptools <65
+  run:
+    - python
+    - {{ pin_compatible('numpy') }}
```

Comment:
Does this work with numpy 2? If so, you should remove this.

Reply:
In setup.py (https://github.com/innolitics/natural-neighbor-interpolation/blob/cccbf97b2fdb709069bf90842445a84acb5de4f8/setup.py#L39), we have:
```
install_requires=[
            'numpy>=1.13',
        ],
```
So I believe it should work with numpy 2 - I've removed this now and just added numpy.

Reply:
BTW, that commit is from last year, long before numpy 2 was out. So there is a chance that it is wrong. The ony way to know this for sure would be to run all the tests with numpy 2.

Reply:
Thanks for the heads-up! I ran the tests, i.e., `/cpptests` and `/tests` hosted on `naturalneighbor` official repository after upgrading `numpy` to version to `>=2.0` in the `setup.py` and all of them passed successfully. The stage-recipe checks also didn't complain when I modified `numpy` to `numpy >=2.0`. If you think these checks are enough, I can push a new commit without version constraint.

Reply:
Unless you used a more updated and/or patched code than the one in this release, I don't think it will pass. This code use deprecate numpy syntax from version 1.20, even before 2.0. Anyway, I added the tests here so we can confirm.

Reply:
As expected:

```
========================= 15 failed, 5 passed in 0.56s =========================
```

You can check the logs for more details, but it is mostly numpy failures due to updates in its API.

Reply:
With the proper numpy pin this is passing now. It will limit it's use, but unless upstream supports modern numpy, there isn't much we can do.